src/functions/contactsMenu/index.ts

```diff
@@ -0,0 +1,47 @@
+/**
+ * SPDX-FileCopyrightText: 2024 Nextcloud GmbH and Nextcloud contributors
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+type ContactsMenuCallback = () => void
+
+interface ContactsMenuHookProps {
+	id: string;
+	callback: ContactsMenuCallback;
+}
+
+declare global {
+	interface Window {
+		_vue_contacts_menu_hooks: Record<string, ContactsMenuHookProps>;
+	}
+}
+
+if (!window._vue_contacts_menu_hooks) {
+	window._vue_contacts_menu_hooks = {}
+}
+
+const registerContactsMenuHook = (id: string, callback: ContactsMenuCallback) => {
```

Comment:
See above about the side effects, just move it to this function will prevent side effects of importing this module so tree shaking can remove all imports if not used and do keep not imports for side effects only.

```suggestion
const registerContactsMenuHook = (id: string, callback: ContactsMenuCallback) => {
    window._vue_contacts_menu_hooks = window._vue_contacts_menu_hooks ?? {}
```

Reply:
Done